VideoWeb/VideoWeb/ClientApp/src/app/services/conference/conference.service.ts

```diff
@@ -0,0 +1,161 @@
+import { ParticipantModel } from 'src/app/shared/models/participant';
+import { ApiClient, ConferenceResponse, ConferenceStatus } from '../clients/api-client';
+import { EventsService } from '../events.service';
+import { LoggerService } from '../logging/logger.service';
+import { ConferenceStatusMessage } from '../models/conference-status-message';
+
+@Injectable({
+    providedIn: 'root'
+})
+export class ConferenceService {
+    private loggerPrefix = '[ConferenceService] -';
+
+    private subscriptions: Subscription[] = [];
+    constructor(
+        router: Router,
+        private activatedRoute: ActivatedRoute,
+        private eventService: EventsService,
+        private apiClient: ApiClient,
+        private logger: LoggerService
+    ) {
+        logger.conferenceService = this;
+        router.events
+            .pipe(
+                filter(x => x instanceof NavigationEnd),
+                map(() => activatedRoute.snapshot),
+                map(route => {
+                    while (route && !route.paramMap?.has('conferenceId')) {
+                        route = route?.firstChild;
+                    }
+
+                    return route?.paramMap;
+                })
+            )
+            .subscribe(paramMap => {
+                this.onRouteParamsChanged(paramMap);
+            });
+    }
+
+    private _currentConference: ConferenceResponse;
+    get currentConference(): ConferenceResponse {
+        return this._currentConference;
+    }
+
+    private currentConferenceSubject = new ReplaySubject<ConferenceResponse>();
+    get currentConference$(): Observable<ConferenceResponse> {
+        return this.currentConferenceSubject.asObservable();
+    }
+
+    private onCurrentConferenceStatusChangedSubject = new BehaviorSubject<{ oldStatus: ConferenceStatus; newStatus: ConferenceStatus }>({
+        oldStatus: this.currentConference?.status,
+        newStatus: null
+    });
+    get onCurrentConferenceStatusChanged$() {
+        return this.onCurrentConferenceStatusChangedSubject.asObservable();
+    }
+
+    private _currentConferenceId: string;
+    get currentConferenceId(): string {
+        return this._currentConferenceId;
+    }
+
+    getConferenceById(conferenceId: string | Guid): Observable<ConferenceResponse> {
+        this.logger.info(`${this.loggerPrefix} getting conference by ID: ${conferenceId}`);
+
+        return this.apiClient.getConferenceById(conferenceId.toString());
+    }
+
+    getParticipantsForConference(conferenceId: Guid | string): Observable<ParticipantModel[]> {
+        this.logger.info(`${this.loggerPrefix} getting participants for conference.`);
+
+        return this.apiClient.getParticipantsByConferenceId(conferenceId.toString()).pipe(
+            map(participants =>
+                participants.map(participantResponse => ParticipantModel.fromParticipantForUserResponse(participantResponse))
+            ),
+            take(1)
```

Comment:
I don't think you need this take(1) as the HttpClient "should" complete the stream after the call. Did you try this without to see?

Reply:
That is a good point. Will remove it.